core/GlobalState.cc

```diff
@@ -1370,7 +1370,7 @@ unique_ptr<GlobalStateHash> GlobalState::hash() const {
     return result;
 }
 
-vector<shared_ptr<File>> GlobalState::getFiles() const {
+const vector<shared_ptr<File>> &GlobalState::getFiles() const {
```

Comment:
this was intentional.
Global state is allowed to relocate the underlying vector and thus taking a reference to it is unsafe.